src/Form/tests/FormAutosuggest.test.jsx

```diff
@@ -25,141 +14,179 @@ function FormAutosuggestWrapper(props) {
   );
 }
 
-describe('FormAutosuggest', () => {
-  afterEach(() => {
-    jest.clearAllMocks();
-  });
-
-  const onSelected = jest.fn();
-  const onClick = jest.fn();
-
-  const container = mount(
+function FormAutosuggestTestComponent(props) {
+  return (
     <FormAutosuggestWrapper
       name="FormAutosuggest"
       floatingLabel="floatingLabel text"
       helpMessage="Example help message"
       errorMessageText="Example error message"
-      onSelected={onSelected}
+      onSelected={props.onSelected}
     >
       <FormAutosuggestOption>Option 1</FormAutosuggestOption>
-      <FormAutosuggestOption onClick={onClick}>Option 2</FormAutosuggestOption>
+      <FormAutosuggestOption onClick={props.onClick}>Option 2</FormAutosuggestOption>
       <FormAutosuggestOption>Learn from more than 160 member universities</FormAutosuggestOption>
-    </FormAutosuggestWrapper>,
+    </FormAutosuggestWrapper>
   );
+}
 
-  describe('render behavior', () => {
-    it('renders component without error', () => {
-      mount(<FormAutosuggestWrapper />);
-    });
+FormAutosuggestTestComponent.defaultProps = {
+  onSelected: jest.fn(),
+  onClick: jest.fn(),
+};
 
-    it('render without loading state', () => {
-      expect(container.exists('.pgn__form-autosuggest__dropdown-loading')).toBe(false);
-      expect(container.props().isLoading).toBeUndefined();
-    });
+FormAutosuggestTestComponent.propTypes = {
+  /** Specifies onSelected event handler. */
+  onSelected: PropTypes.func,
+  /** Specifies onClick event handler. */
+  onClick: PropTypes.func,
+};
 
-    it('render with loading state', () => {
-      const wrapper = mount(<FormAutosuggestWrapper isLoading />);
+describe('render behavior', () => {
+  it('renders component without error', () => {
+    render(<FormAutosuggestWrapper />);
+  });
 
-      expect(wrapper.exists('.pgn__form-autosuggest__dropdown-loading')).toBe(true);
-      expect(wrapper.props().isLoading).toBe(true);
-    });
+  it('renders without loading state', () => {
+    const { container } = render(<FormAutosuggestTestComponent />);
+    expect(container.querySelector('.pgn__form-autosuggest__dropdown-loading')).toBeNull();
+  });
 
-    it('renders the auto-populated value if it exists', () => {
-      const wrapper = mount(<FormAutosuggestWrapper value="Test Value" />);
+  it('render with loading state', () => {
+    const { container } = render(<FormAutosuggestWrapper isLoading />);
+    expect(container.querySelector('.pgn__form-autosuggest__dropdown-loading')).toBeTruthy();
```

Comment:
Similar to above comment, could we either assert that the `screenReaderText` passed to the `Spinner` is rendered, or alternatively, rely on a more stable `data-testid` instead of the class name?

Reply:
Fixed in:
https://github.com/openedx/paragon/compare/299441986865ab87377629aa4d99f13ed8fc47d7..6c9a96ea3e2ba2ef4ab37d5f266b4d7205f5c718